Integration Test Target Setup

aashishpatil-g opened the pull request:
Adds an Integration Test Target. 

Tests cannot be run within a git action as yet but I've run them manually locally and they are executing successfully.

paulb777 commented:
Looks like copyrights are getting deleted

aashishpatil-g commented:
> Looks like copyrights are getting deleted

Sorry about that. Missed them in the generated code. Added now.

paulb777 commented:
Consider using an environment variable like in #8 to wrap integration tests so CI can succeed.  

I'll investigate working around the CI issue in another PR with less parallelization

aashishpatil-g commented:
> Consider using an environment variable like in #8 to wrap integration tests so CI can succeed.
> 
> I'll investigate working around the CI issue in another PR with less parallelization

Added a flag to disable integration tests and added this flag to the spm.yml

paulb777 commented on Tests/Integration/Gen/KitchenSink/Sources/KitchenSinkClient.swift:
```diff
@@ -15,15 +15,13 @@
 import FirebaseDataConnect
 import Foundation
 
-@available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
```
Was it intentional to remove these? Shouldn't the Package.swift versions be updated to these in conjunction with this removal?

aashishpatil-g replied:
The style script seems to have removed these. They are part of the generated code.

paulb777 replied:
Hmm, the style script isn't touching those when I run it ...

aashishpatil-g replied:
Sorry, I think they got stripped out when I regenerated the code. The code is not part of the core SDK. Its generated so we can run the integration tests.

aashishpatil-g replied:
We have the version gates on the tests and core sdk so the gate on the gen code isn't required.
